tasks/Task.cpp

```diff
@@ -167,4 +196,238 @@ void Task::processPointcloudData(LivoxLidarEthernetPacket const* data)
             m_measurements_merged = 0;
         }
     }
-}
\ No newline at end of file
+
+    // TODO CARTESIANCOORDINATELOWDATA
+    // TODO SPHERICALCOORDINATEDATA
+}
+
+void Task::notifyCommandSuccess()
+{
+    unique_lock<std::mutex> lock(m_command_sync_mutex);
+    m_error_code = 0;
+    m_command_sync_condition.notify_all();
+}
+
+void Task::notifyCommandFailure(int error_code)
+{
+    unique_lock<std::mutex> lock(m_command_sync_mutex);
+    m_error_code = error_code;
+    m_command_sync_condition.notify_all();
+}
+
+void Task::waitForCommandSuccess()
+{
+    unique_lock<std::mutex> lock(m_command_sync_mutex);
+
+    m_command_sync_condition.wait(lock);
+    if (m_error_code) {
+        throw std::runtime_error(m_lidar_status[m_error_code]);
```

Comment:
use .at() instead of [] to do bounds checks (I think it is hard to guarantee that m_error_code won't ever get out of m_lidar_status)